crates/polars-plan/src/plans/optimizer/slice_pushdown_lp.rs

```diff
@@ -496,15 +496,16 @@ impl SlicePushDown {
             // [Pushdown]
             // these nodes will be pushed down.
             // State is None, we can continue
-            m @(Select {..}, None) |
-            m @ (SimpleProjection {..}, _)
+            m @ (Select {..}, None)
+            | m @ (HStack {..}, None)
```

Comment:
The original MRE of the panic didn't actually contain a `slice()` - I realized this was because we were always checking the projections even when we didn't need to.